src/SafeAreaContext.tsx

```diff
@@ -31,78 +31,101 @@ export interface SafeAreaProviderProps extends ViewProps {
   initialSafeAreaInsets?: EdgeInsets | null;
 }
 
-export function SafeAreaProvider({
-  children,
-  initialMetrics,
-  initialSafeAreaInsets,
-  style,
-  ...others
-}: SafeAreaProviderProps) {
-  const parentInsets = useParentSafeAreaInsets();
-  const parentFrame = useParentSafeAreaFrame();
-  const [insets, setInsets] = React.useState<EdgeInsets | null>(
-    initialMetrics?.insets ?? initialSafeAreaInsets ?? parentInsets ?? null,
-  );
-  const [frame, setFrame] = React.useState<Rect>(
-    initialMetrics?.frame ??
-      parentFrame ?? {
-        // Backwards compat so we render anyway if we don't have frame.
-        x: 0,
-        y: 0,
-        width: Dimensions.get('window').width,
-        height: Dimensions.get('window').height,
+export const SafeAreaProvider = React.memo(
```

Comment:
`memo` on the entire component, I'm not too sure if we need this, but went all the way.

Reply:
I think in general components that accept children can never really be memo'd, since the children basically always change

Reply:
Great point, will remove it.

Reply:
Pushed up the change.